packages/nvidia_nat_adk/src/nat/plugins/adk/adk_callback_handler.py

```
@@ -140,12 +140,17 @@ async def wrapped_tool_use(base_tool_instance, *args, **kwargs) -> Any:
 
+                # Store the UUID to ensure the END event uses the same ID
+                step_uuid = stats.UUID
                 self.step_manager.push_intermediate_step(stats)
 
                 with self._lock:
                     self.last_call_ts = now
 
                 # Call the original _use(...)
+                if original_func is None:
+                    raise RuntimeError(
+                        "Original tool function is None - instrumentation may not have been set up correctly")
```

Ruff (0.13.1)

Abstract `raise` to an inner function

(TRY301)